todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/providers/BaseNavigationProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.example.android.architecture.blueprints.todoapp.util.providers;
+
+/**
+ * Handles navigation between Activities in the app.
+ */
+public interface BaseNavigationProvider {
```

Comment:
I would call all the "navigation providers" simply "navigators" which is the known pattern name.